lib/simulator/SimulatorLightController.cpp

```diff
@@ -10,92 +8,58 @@
 #include <utility>
 #include <vector>
 
-SimulatorLightController::SimulatorLightController(Ogre::SceneManager *scnMgr)
-    : scnMgr(scnMgr) {
-  for (int pole_index = 0; pole_index < 6; pole_index++) {
-    auto pole = Pole();
-    const unsigned long grid_num_rows = pole.get_grid_lights().size();
-    const unsigned long grid_num_cols = pole.get_grid_lights().front().size();
-
-    std::vector<std::vector<Ogre::Light *>> light_pole;
-    for (unsigned long row = 0; row < grid_num_rows; row++) {
-      std::vector<Ogre::Light *> light_row;
-      for (unsigned long col = 0; col < grid_num_cols; col++) {
-        Ogre::Vector3 unscaled = Ogre::Vector3(
-            kCircleRadiusInches * cos(M_PI / 3.0 * pole_index) +
-                col * kPixelPitchInches * sin(M_PI / 3.0 * pole_index),
-            kPoleHeightInches - row * kPixelPitchInches,
-            kCircleRadiusInches * sin(M_PI / 3.0 * pole_index) +
-                col * kPixelPitchInches * cos(M_PI / 3.0 * pole_index));
-
-        Ogre::Vector3 scaled =
-            Ogre::Vector3(SimulatorLightController::inchesToCoords(unscaled.x),
-                          SimulatorLightController::inchesToCoords(unscaled.y),
-                          SimulatorLightController::inchesToCoords(unscaled.z));
-        light_row.push_back(createLight(scaled));
-      }
-      light_pole.push_back(std::move(light_row));
-    }
-    poles.push_back(std::move(pole));
-    lights.push_back(std::move(light_pole));
-  }
+SimulatorLightController::SimulatorLightController() {
+  led_pixels_ = kLedPixels;
+  led_frame_pixels_ = kLedFramePixels;
 }
 
 void SimulatorLightController::WriteOutLights() {
-  for (int pole_index = 0; pole_index < 6; pole_index++) {
-    auto &grid_lights = poles[pole_index].get_grid_lights();
-    const uint8_t grid_num_rows = grid_lights.size();
-    const uint8_t grid_num_cols = grid_lights.front().size();
+  for (uint8_t pole_number = 0; pole_number < Pole::kNumPoles; pole_number++) {
+    auto& grid_lights = poles[pole_number].get_grid_lights();
 
-    for (unsigned long x = 0; x < grid_num_rows; x++) {
-      for (unsigned long y = 0; y < grid_num_cols; y++) {
-        lights[pole_index][x][y]->setSpecularColour(
-            grid_lights[x][y].r / 255.0 * kLightScale,
-            grid_lights[x][y].g / 255.0 * kLightScale,
-            grid_lights[x][y].b / 255.0 * kLightScale);
-        lights[pole_index][x][y]->setDiffuseColour(
-            grid_lights[x][y].r / 255.0 * kLightScale,
-            grid_lights[x][y].g / 255.0 * kLightScale,
-            grid_lights[x][y].b / 255.0 * kLightScale);
-      }
-    }
-  }
-}
+    leds[pole_number * kLedsPerOutput + 0] = grid_lights[0][0];
```

Comment:
It looks like this entire loop body could be simplified to

```
auto& grid_lights = poles[pole_number].get_grid_lights();
for (std::size_t n = 0; n != 16; ++n) {
  leds[pole_number * kLedsPerOutput + n] = grid_lights[n / 4][n % 4];
}
```

Reply:
And then it would be nice if `16` weren't some magic number.

Reply:
Done.